src/braket/circuits/measure.py

```diff
@@ -0,0 +1,109 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+from typing import Any, Optional
+
+from braket.circuits.quantum_operator import QuantumOperator
+from braket.circuits.serialization import (
+    IRType,
+    OpenQASMSerializationProperties,
+    SerializationProperties,
+)
+from braket.registers.qubit_set import QubitSet
+
+
+class Measure(QuantumOperator):
+    """Class `Measure` represents a measure operation on targeted qubits"""
+
+    def __init__(
+        self, qubit_count: Optional[int] = 1, ascii_symbols: Sequence[str] = ["M"], **kwargs
+    ):
+        """Inits a `Measure`.
+
+        Args:
+            qubit_count (Optional[int]): Number of qubits this measure operation interacts with.
```

Comment:
I'm not sure about the qubit count for this op. Seems like it might just always be one, unless we want to  represent measuring multiple qubits at once as physically different than one at a time.

If we do have this arg, then it should likely be part of the `__eq__` comparison

Reply:
Added `qubit_count` to the `__eq__` comparison.

The `qubit_count` needs to match the number of targets and if the `measure` instruction takes in a list of targets like: `Instruction(Measure(qubit_count=3), [0, ,1 , 2])` then Measure would need a qubit count of 3

Reply:
I also am not sure whether `qubit_count` should be here. It looks like we always serialize into single-qubit measure instructions in the OpenQASM anyway. (Note that we also do the same today with the implicit all-qubit measurement, so this is consistent - which is good.)

So it's just a question of whether we want the Circuit object to retain a difference between `circ.measure([0, 1])` vs. `circ.measure(0).measure(1)`. Since there's no difference in the resulting program, it feels like these should be treated the same. So personally I think it makes the most sense to treat this as a single-qubit instruction and have `circ.measure([0, 1])` be equivalent to `circ.measure(0).measure(1)` -- just like `circ.h([0, 1])` is equivalent to `circ.h(0).h(1)`.

But I don't think this is a very critical decision.

Reply:
Those are good points. I removed `qubit_count` from here to be more consistent.